distributed/shuffle/_rechunk.py

```diff
@@ -410,9 +414,15 @@ def _shard_partition(
             if shard.base is not None:
                 shard = shard.copy()
 
+            shards_size += shard.nbytes
+            shards_count += 1
+
             out[self.worker_for[chunk_index]].append(
                 (chunk_index, (shard_index, shard))
             )
+
+        context_meter.digest_metric("shuffle-shards", shards_size, "bytes")
+        context_meter.digest_metric("shuffle-shards", shards_count, "count")
```

Comment:
bytes / count = mean shard size

Reply:
Let's look at performance for these in-depth metrics, I'm not sure if collecting shard size here is _always_ useful, so we may want a toggle.